components/modals/EditTaskModal.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import moment from 'moment';
+import { NotificationDateTimeType, TaskData, TaskFormData, TaskPatchData } from '../../types/task';
+import { isValidDate } from '../../utils/date/validations';
+import { calculateDaysOffset } from '../../utils/date/formatters';
+import { Nullable } from '../../types/utils';
+import TaskForm from '../forms/TaskForm';
+import tasksApi from '../../api/tasksApi';
+import {
+  DEFAULT_NOTIFICATION_DAYS_OFFSET,
+  DEFAULT_NOTIFICATION_TIME,
+  getNotificationDateTime,
+} from '../../utils/task/taskUtils';
+import { Button, Modal } from 'antd';
+import { ExclamationCircleOutlined } from '@ant-design/icons';
+
+const { confirm } = Modal;
+
+interface Props {
+  applicationId: number;
+  initialTask: TaskData;
+  setSelectedTask: React.Dispatch<React.SetStateAction<Nullable<TaskData>>>;
+  setShouldFetchData: Dispatch<SetStateAction<boolean>>;
+}
+
+function EditTaskModal({ applicationId, initialTask, setSelectedTask, setShouldFetchData }: Props) {
+  const [task, setTask] = useState(initialTask);
+  const [initialValues, setInitialValues] = useState<TaskFormData>({
+    notificationDaysOffset: DEFAULT_NOTIFICATION_DAYS_OFFSET,
+  });
+  const [shouldDisableSaveButton, setShouldDisableSaveButton] = useState(true);
```

Comment:
Let's standarize to use generics for inferable types.

But this is up to you to decide.

This applies to all other files as well.